quic/s2n-quic-events/events/common.rs

```diff
@@ -650,6 +650,41 @@ enum PacketDropReason<'a> {
     },
 }
 
+enum AckAction {
+    /// Ack range for received packets was dropped due to space constraints
+    ///
+    /// For the purpose of processing Acks, RX packet numbers are stored as
+    /// packet_number ranges in an IntervalSet; only lower and upper bounds
+    /// are stored instead of individual packet_numbers. Ranges are merged
+    /// when possible so only disjointed ranges are stored.
+    ///
+    /// When at `capacity`, the lowest packet_number range is dropped.
+    RxAckRangeDropped {
+        /// Start of the packet number range which was dropped
+        range_start: u64,
+        /// End of the packet number range which was dropped
+        range_end: u64,
+        /// The number of disjoint ranges the IntervalSet can store
+        capacity: u16,
+        /// The min value store in the IntervalSet
+        stored_min: u64,
+        /// The max value stored in the IntervalSet
+        stored_max: u64,
+    },
```

Comment:
I was considering if instead of `range_start` and `range_end` should this be `range: core::ops::Range`?

but want to think if [we really want Range](https://kaylynn.gay/blog/post/rust_ranges_and_suffering). Leaning towards maybe our own newtype which we can then expose functions on.

Reply:
exposed a newtype `Range` in https://github.com/aws/s2n-quic/pull/1314/commits/3ced090dceb9ba21bd9b66de39f3b5238c69bd0c